src/wasm_worker.js

```diff
@@ -28,52 +27,44 @@ if (ENVIRONMENT_IS_NODE) {
     return f;
   }
 
-  var fs = require('fs');
-  var vm = require('vm');
-
-  Object.assign(global, {
+  Object.assign(globalThis, {
     self: global,
-    require,
-#if !EXPORT_ES6
-    // `vm.runInThisContext` global scope lacks `__filename` and `__dirname`
-    __filename,
-    __dirname,
-#endif
-    Worker: nodeWorkerThreads.Worker,
-    importScripts: (f) => vm.runInThisContext(fs.readFileSync(f, 'utf8'), {filename: f}),
     postMessage: (msg) => parentPort.postMessage(msg),
```

Comment:
I think these polyfills could be shared with runtime_pthread too: https://github.com/emscripten-core/emscripten/blob/081ac97000e8c609db809df50672b8d5a04e545e/src/runtime_pthread.js#L29-L35 + https://github.com/emscripten-core/emscripten/blob/f163dc219468cf98873a054319be19aacfec5a79/src/wasm_worker.js#L17

Reply:
True. Refactored shared code.